Add example to docs

casey opened the pull request:
Here's an example of the type of example that would have helped me figure out how to use darling, see #197 

- Shows the derive macro input
- Shows the darling parser
- Extracts a single top-level attribute

A sequence of examples such as these, each one performing a single task, and exercising a single feature of darling, would make it easy to get started.

I think that doctests are the ideal place to put these, for discoverability and ease of browsing, but there are multiple difficulties:

- Doctests cannot use #[proc_macro_derive]
- Doctests can't contain both the derive invocation and parser

But I wanted to open this PR as a starting point for discussion.

TedDriggs commented on src/lib.rs:
```diff
@@ -63,6 +65,40 @@
+//! pub fn display(item: proc_macro::TokenStream) -> proc_macro::TokenStream {
+//!   let derive_input = syn::parse_macro_input!(item as syn::DeriveInput);
+//!
+//!   let foo = Foo::from_derive_input(&derive_input).unwrap();
```
We shouldn't use `unwrap` in an example. There's a right way to check the errors which will cause them to be emitted as compile errors; I have to look for where it's demonstrated, since it's been a while since I looked at this piece of the code.

TedDriggs replied:
[Here is a better approach from `derive_builder_core`](https://github.com/colin-kiegel/rust-derive-builder/blob/master/derive_builder_core/src/lib.rs#L62-L67)

```rust
    let opts = match macro_options::Options::from_derive_input(&ast) {
        Ok(val) => val,
        Err(err) => {
            return err.write_errors();
        }
    };
```

The `write_errors()` call produces a `proc_macro2::TokenStream`, so calling `into` on it will then convert it into the `proc_macro::TokenStream` expected by the derive entry-point function.

TedDriggs commented on src/lib.rs:
```diff
@@ -1,8 +1,9 @@
 //! # Darling
-//! Darling is a tool for declarative attribute parsing in proc macro implementations.
 //!
+//! Darling is a tool for declarative attribute parsing in proc macro implementations.
```
Are these whitespace changes necessary?

casey replied:
No, but they do help make the doc comment more readable in an editor. I can remove them if you'd prefer.

TedDriggs reviewed (changes requested):
Thanks for improving this! I've left a few comments; the error handling is the biggest one; I'm trying to find a great example of the right way to do it.

casey commented on src/lib.rs:
````diff
@@ -63,6 +65,40 @@
+//! ```ignore
+//! #[derive(Foo)]
+//! #[foo(bar = "hello")]
+//! struct Foo {
````
I actually think `Foo`, `foo`, and `bar` are useful here. Darling assigns special meaning to a few names, and this makes it clear that these are names which are up to the user to decide. I did go ahead and rename `struct Foo` to `struct Input`.

casey commented:
Added some commits addressing your comments. I didn't remove the whitespace changes, they're just to improve readability, but I can undo those or put them in another PR.

casey commented:
@TedDriggs Friendly ping!

TedDriggs commented:
Apologies for the delay - I'll be busy with work for the next few days, but haven't forgotten about this and will get back to it as soon as I can.
